clang/lib/Serialization/ASTWriter.cpp

```diff
@@ -7230,6 +7230,9 @@ void ASTWriter::CompletedImplicitDefinition(const FunctionDecl *D) {
   if (!D->isFromASTFile())
     return; // Declaration not imported from PCH.
 
+  if (!D->doesThisDeclarationHaveABody())
+    return; // The function definition may not have a body due to parsing errors.
```

Comment:
I prefer to see the comments above the if.

Reply:
Moved